nautobot/core/api/utils.py

```diff
@@ -335,7 +335,9 @@ def return_nested_serializer_data_based_on_depth(serializer, depth, obj, obj_rel
     if obj_related_field.__class__.__name__ == "RelatedManager":
         result = []
         if depth == 0:
-            result = obj_related_field.values_list("pk", flat=True)
+            result = [entry.get_absolute_url(api=True) for entry in obj_related_field.all()]
+            if serializer.context.get("request"):
+                result = [serializer.context.get("request").build_absolute_uri(url) for url in result]
```

Comment:
Why the distinction here with `request` vs `no request`? Is `get_absolute_url` here not enough to make the BrowableAPI walkable?

Reply:
It's for self-consistency. `get_absolute_url` alone doesn't know about the protocol/host/port info (it returns e.g. `/api/dcim/devices/<uuid>`), `request.build_absolute_uri` is what adds that information (`http://localhost:3000/api/dcim/devices/<uuid>`).